### PR TITLE
[feat][Dock] Add information about Dock spring-loading option

### PR DESCRIPTION
**Pull request by jidicula**
**Why this change was necessary**
There wasn't an existing entry in the site about Dock spring-loading.

**What this change does**
Adds an entry into the site for Dock icon spring loading for folders
and applications.

**Any side-effects?**
None.

**Additional context/notes/links**

Resolves yannbertrand/macos-defaults#51

See also: https://www.defaults-write.com/enable-spring-loaded-dock-items

### CODE REVIEW

**Comment by jidicula** [comment]
@yannbertrand I don't have the ability to test this on Big Sur, so it would be great if you or someone else could verify that this works there.

**Comment by yannbertrand** [comment]
Hey @jidicula, thanks for the contribution!

I just tried it on Big Sur, it's seems I can't do it anymore 😭 

Can you show me a little video on Catalina on what's expected of each options to make sure I do it right? You can record using `cmd`+`shift`+`5`

File: defaults.yml [hunk 1]
```diff
@@ -171,6 +171,20 @@ categories:
             name: com.apple.dock autohide
             value: true
         after: killall Dock
+      - key: enable-spring-load-actions-on-all-items # The command's key
+        domain: com.apple.dock # The command's domain
```
Comment (yannbertrand) [inline defaults.yml:175]:
```suggestion
      - key: enable-spring-load-actions-on-all-items
        domain: com.apple.dock
```

**Comment by jidicula** [comment]
@yannbertrand Ok, I made a recording for when the option is true and when it is false. Sorry that it's a zipfile, GitHub didn't allow me to upload .mov.
[recordings.zip](https://github.com/yannbertrand/macos-defaults/files/5322023/recordings.zip)

Now that I've made the recording, it looks like the default behaviour of `false` is very similar to the `true` behaviour – the only difference I see in the `true` case is that Preview opens as an application, then opens the file when you drop it on the icon, while in the default `false` case, Preview doesn't open as an application until you release the click after hovering. I'm not sure if this difference in behaviour actually changes anything for the user, so maybe this shouldn't be added to the list of commands?

**Comment by yannbertrand** [comment]
Thanks for the videos.

Alright I think understand the difference and I got something similar working on Big Sur. A small difference is that it doesn't open the file itself but the File opening selector. I got the same behaviour on Catalina 10.15.6, may we have another config difference?
set to true while Preview closed
If Preview is already opened, I don't notice any difference between setting true or false.

Did you notice any change on other apps than Preview or QuickTime Player tho?

**Comment by jidicula** [comment]
I just tried it with Keynote and it's exactly the same as with Preview in your gif – it opens the file dialog when the setting is true. When the setting is false, no file dialog opens. With both settings, releasing the click opens the .key file in Keynote.

**Comment by yannbertrand** [comment]
The time we took looking for what the command do is valuable, I think it may be useful for other people. What about adding some info and moving the command to the miscellaneous folder?

**Comment by jidicula** [comment]
Sure, I can move it to the miscellaneous folder. What other info should I include?

**Comment by yannbertrand** [comment]
Perfect. I'd say we should explain that the behaviour is different if the App is already opened or not. And precise that we've seen it working on Preview/QuickTime Player/Keynote.

File: defaults.yml [hunk 2]
```diff
@@ -459,4 +459,23 @@ categories:
             text: By default, the Help Menu is always-on-top
           - value: true
             text: The Help Menu can go behind other windows
+
```
Comment (yannbertrand) [inline defaults.yml:462]:
```suggestion
```

**Comment by yannbertrand** [comment]
Btw have you seen it working on a folder?

**Comment by jidicula** [comment]
> Btw have you seen it working on a folder?

The behaviour doesn't change with a folder between `true` and `false`.

File: defaults.yml [hunk 3]
```diff
@@ -460,3 +460,21 @@ categories:
...
+          Quicktime, and iWork (Keynote, Pages, Numbers).
+        param:
+          type: bool
+        examples:
+          - value: false
+            default: true
+            text: Spring loading is disabled for Dock items.
+          - value: true
+            text: Spring loading is enabled for Dock items.
+        versions: [Catalina]
```
Comment (yannbertrand) [inline defaults.yml:479]:
```suggestion
        versions: [Big Sur, Catalina]
```

**Comment by yannbertrand** [comment]
LGTM! Thanks @jidicula 🙂

**Comment by jidicula** [comment]
My pleasure, glad to contribute 😁

**Pull request merged**